File: clients/python/auth_client.py
"""
Authentication Client

:author Yepeng Ding
"""
import requests


class AuthClient:
    """
    Authentication Client contains APIs to interact with BoostAuth.
    """

    def __init__(self, domain: str):
        """
        Construct a new 'AuthClient' object.

        :param domain: BoostAuth domain (e.g., boostauth.org)
        """
        self.domain = domain

    def login(self, username: str, password: str, source: str) -> dict:
        """
        Login interface.

        :param username: username
        :param password: password
        :param source: source
        :return: JSON response
        """
        return requests.post(f'{self.domain}/auth/login', json={
            "username": username,
            "password": password,
            "source": source}).json()

    def logout(self, token: str) -> dict:
        """
        Logout interface.

        :param token: token obtained after login
        :return: JSON response
        """
        return requests.post(f'{self.domain}/auth/logout', json={
            "token": token}).json()

    def verify(self, token: str) -> dict:
        """
        Verify interface.

        :param token: token obtained after login
        :return: JSON response
        """
        return requests.post(f'{self.domain}/verify', json={
            "token": token}).json()

File: clients/php/AuthClient.php
<?php

/**
 * Authentication Client.
 * PHP 7.4+
 *
 * @author Yepeng Ding
 */
class AuthClient
{
    /**
     * @var string $domain
     */
    private string $domain;

    /**
     * Construct a new 'AuthClient' object.
     *
     * @param string $domain BoostAuth domain (e.g., boostauth.org)
     */
    public function __construct(string $domain)
    {
        $this->domain = $domain;
    }

    /**
     * Login interface.
     *
     * @param string $username username
     * @param string $password password
     * @param string $source source
     * @return mixed PHP object
     */
    public function login(string $username, string $password, string $source): mixed
    {
        return $this->post($this->domain . '/auth/login', array(
            'username' => $username,
            'password' => $password,
            'source' => $source
        ));
    }

    /**
     * Logout interface.
     *
     * @param string $token token obtained after login
     * @return mixed PHP object
     */
    public function logout(string $token): mixed
    {
        return $this->post($this->domain . '/auth/logout', array(
            'token' => $token
        ));
    }

    /**
     * Verify token interface.
     *
     * @param string $token token obtained after login
     * @return mixed PHP object
     */
    public function verify(string $token): mixed
    {
        return $this->post($this->domain . '/verify', array(
            'token' => $token
        ));
    }

    /**
     * Post method.
     *
     * @param string $url url
     * @param array $data request payload
     * @return mixed PHP object
     */
    private function post(string $url, array $data): mixed
    {
        $options = array(
            'http' => array(
                'header' => "Content-Type: application/json\r\n" . "Accept: application/json\r\n",
                'method' => 'POST',
                'content' => json_encode($data)
            )
        );

        $context = stream_context_create($options);
        $result = file_get_contents($url, false, $context);

        return json_decode($result);
    }
}